pkg/build/admission/buildpodutil.go

```diff
@@ -0,0 +1,69 @@
+package admission
+
+import (
+	"errors"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+func IsBuildPod(pod *kapi.Pod) bool {
+	return hasBuildAnnotation(pod) && hasBuildEnvVar(pod)
+}
+
+func GetBuildFromPod(pod *kapi.Pod, codec runtime.Codec) (*buildapi.Build, error) {
+	envVar, err := buildEnvVar(pod)
+	if err != nil {
+		return nil, err
+	}
+	build := &buildapi.Build{}
+	err = codec.DecodeInto([]byte(envVar.Value), build)
+	if err != nil {
+		return nil, err
+	}
+	return build, nil
+}
+
+func SetBuildInPod(build *buildapi.Build, pod *kapi.Pod, codec runtime.Codec) error {
+	envVar, err := buildEnvVar(pod)
+	if err != nil {
+		return err
+	}
+	encodedBuild, err := codec.Encode(build)
```

Comment:
You need to be sure that you re-encode using the same external group/version that passed to you.  There's a method in scheme called `DataKind` that can give you that information from the bytes